### PR TITLE
Fix Error if Non-existent Storage Account Specified

### LINKED ISSUE #784: Validation error on account name & account key parameters
Set account name but not key.
```
$ export AZURE_STORAGE_ACCOUNT=mystorageaccount
```
After running `$ az storage blob list -c vhds`, I expect an error saying storage key not specified or similar but instead, I get the following.

```
$ az storage blob list -c vhds
usage: az storage blob list [-h] [--output {json,tsv,list,table,jsonc}]
                            [--verbose] [--debug] [--query JMESPATH]
                            --container-name CONTAINER_NAME
                            [--timeout TIMEOUT] [--account-name ACCOUNT_NAME]
                            [--marker MARKER] [--sas-token SAS_TOKEN]
                            [--connection-string CONNECTION_STRING]
                            [--num-results NUM_RESULTS] [--include INCLUDE]
                            [--delimiter DELIMITER]
                            [--account-key ACCOUNT_KEY] [--prefix PREFIX]
az storage blob list: error: 'NoneType' object has no attribute 'id'
```

The bug is [in _validators.py](https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py#L46), `acc_id = next((x for x in scf.storage_accounts.list() if x.name == n.account_name), None).id`.

If n.account_name isn't in the storage accounts list, the iterator will return None and so `'NoneType' object has no attribute 'id'`

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes #784.

### CODE REVIEW

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py [hunk 1]
```diff
@@ -43,12 +43,14 @@ def validate_client_parameters(namespace):
     # if account name is specified but no key, attempt to query
     if n.account_name and not n.account_key:
         scf = get_mgmt_service_client(StorageManagementClient)
-        acc_id = next((x for x in scf.storage_accounts.list() if x.name == n.account_name), None).id
+        acc_id = next((x for x in scf.storage_accounts.list() if x.name == n.account_name), None)
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py]: Can we rename `acc_id` to `acc` now as it's actually the account that's returned.

Comment (tjprescott) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py]: Anything for you Derek :D

---
In reply to: [76833036](https://github.com/Azure/azure-cli/pull/787#discussion_r76833036) [](ancestors = 76833036)

**Comment by tjprescott** [comment]
@derekbekoe comments fixed.

**Comment by derekbekoe** [comment]
:shipit:

**Pull request merged**